packages/terra-clinical-data-grid/src/subcomponents/HeaderCell.jsx

```diff
@@ -0,0 +1,132 @@
+   * String identifier of the column in which the HeaderCell will be rendered.
+   */
+  columnId: PropTypes.string.isRequired,
+  /**
+   * String-formatted width that the HeaderCell should be rendered as. Any valid css width value is supported (i.e. 200px, 3rem).
+   */
+  width: PropTypes.string.isRequired,
+  /**
+   * String text that will be rendered within the HeaderCell's default text region.
+   */
+  text: PropTypes.string,
+  /**
+   * String indicating the sort direction of the column. If provided, a sorting indicator will be
+   * rendered within in the header. However, if custom children are also provided, the sorting indicator will
+   * not be rendered.
+   */
+  sortIndicator: PropTypes.oneOf(['ascending', 'descending']),
+  /**
+   * Boolean indicating whether or not the HeaderCell is selectable.
+   */
+  isSelectable: PropTypes.bool,
+  /**
+   * Function that will be called when the HeaderCell is selected. The `isSelectable` prop must be true for
+   * this function to be called.
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Boolean indicating whether or not the HeaderCell is resizable. If true, a ResizeHandle will be rendered
+   * over the HeaderCell's contents.
+   */
+  isResizable: PropTypes.bool,
+  /**
+   * Function that will be called upon the release of the ResizeHandle. The `isResizable` prop must be be true
+   * for this function to be called.
+   */
+  onResizeEnd: PropTypes.func,
+  /**
+   * Content that will rendered within the HeaderCell. If provided, the `text` and `sortIndicator` props will be ignored.
+   */
+  children: PropTypes.node,
+  /**
+   * Function that will be called with a ref to the Cell's selectable element.
+   */
+  selectableRefCallback: PropTypes.func,
+};
+
+class HeaderCell extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleCellSelect = this.handleCellSelect.bind(this);
+  }
+
+  handleCellSelect(sectionId, rowId, columnId) {
+    const { onSelect } = this.props;
+
+    if (onSelect) {
+      onSelect(columnId);
+    }
+  }
+
+  render() {
+    const {
+      columnId,
+      text,
+      sortIndicator,
+      isSelectable,
+      isResizable,
+      width,
+      children,
+      onResizeEnd,
+      selectableRefCallback,
+    } = this.props;
+
+    let content = children;
+    if (!content && (text || sortIndicator)) {
+      let sortIndicatorComponent;
+      if (sortIndicator) {
+        sortIndicatorComponent = (
+          <div className={cx('sort-indicator')}>
+            {sortIndicator === 'ascending' ? <IconCaretUp /> : null}
+            {sortIndicator === 'descending' ? <IconCaretDown /> : null}
+          </div>
+        );
+      }
+
+      content = (
+        <div className={cx('default-layout')}>
+          <div className={cx('text')}>
+            {text}
+          </div>
+          {sortIndicatorComponent}
+        </div>
+      );
+    }
+
+    return (
+      <Cell
+        className={cx('header-cell')}
+        sectionId=""
```

Comment:
I feel like we would want a sectionID and rowID? Or do we have a default Header value we should add?